src/vcSettings.cpp

```diff
@@ -250,14 +250,48 @@ bool vcSettings_Load(vcSettings *pSettings, bool forceReset /*= false*/, vcSetti
     pSettings->maptiles.mapEnabled = data.Get("maptiles.enabled").AsBool(true);
     pSettings->maptiles.demEnabled = data.Get("maptiles.demEnabled").AsBool(true);
 
-    udStrcpy(pSettings->maptiles.mapType, data.Get("maptiles.type").AsString("euc-az-aerial"));
+    pSettings->maptiles.activeLayerCount = data.Get("maptiles.layerCount").AsInt(0);
 
-    udStrcpy(pSettings->maptiles.customServer.tileServerAddress, data.Get("maptiles.serverURL").AsString("https://slippy.vault.euclideon.com/{0}/{1}/{2}.png"));
-    udStrcpy(pSettings->maptiles.customServer.attribution, data.Get("maptiles.attribution").AsString("\xC2\xA9 OpenStreetMap contributors"));
+    // backwards compatability
+    if (pSettings->maptiles.activeLayerCount == 0)
+    {
+      // defaults
+      pSettings->maptiles.activeLayerCount = 1;
+      pSettings->maptiles.layers[0].enabled = true;
+      for (int mapLayer = 1; mapLayer < vcMaxTileLayerCount; ++mapLayer)
+      {
+        pSettings->maptiles.layers[mapLayer].mapHeight = 0.0f;
+        pSettings->maptiles.layers[mapLayer].blendMode = vcMTBM_Hybrid;
+        pSettings->maptiles.layers[mapLayer].transparency = 1.0f;
+      }
+
+      udStrcpy(pSettings->maptiles.layers[0].mapType, data.Get("maptiles.layer[0].type").AsString("euc-az-aerial"));
```

Comment:
This is the backwards compatibility path, these `data.Get()` calls shouldn't contain `.layer[0]`.

Reply:
Good catch. Yes this was very broken.